### PR TITLE
feat(datasource/cpan): populate latest tag

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
Use the [module/{module} ](https://github.com/metacpan/metacpan-api/blob/master/docs/API-docs.md#modulemodule)endpoint to get the latest version and populate `tags[latest]` field

## Context
Ref: https://github.com/renovatebot/renovate/discussions/32670#discussioncomment-11348627

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository [Before](https://github.com/wwuck/renovate-cpanfile-itch/pull/1)  ..... [After](https://github.com/Rahul-renovate-testing/renovate-cpanfile-itch) (no pr created as we are on the latest version)

### CODE REVIEW

File: lib/modules/datasource/cpan/index.ts [hunk 1]
```diff
@@ -107,8 +107,24 @@ export class CpanDatasource extends Datasource {
...
+        ).body;
+        const latestVersion = latestVersionRes.module[0].version;
+
+        if (latestVersion) {
+          result.tags ??= {};
+          result.tags.latest ??= latestVersion;
+        }
       }
     }
+
```
Comment (RahulGautamSingh) [inline lib/modules/datasource/cpan/index.ts:127]:
```suggestion
```

File: lib/modules/datasource/cpan/index.ts [hunk 2]
```diff
@@ -107,6 +107,21 @@ export class CpanDatasource extends Datasource {
...
+        const latestVersionUrl = joinUrlParts(
+          registryUrl,
+          'v1/module/',
+          packageName,
+        );
+        const latestVersionRes = (
+          await this.http.getJson<MetaCpanApiFile>(latestVersionUrl)
+        ).body;
+        const latestVersion = latestVersionRes.module[0].version;
```
Comment (rarkins) [inline lib/modules/datasource/cpan/index.ts:119]: This needs to be more "defensive" in case data is not as we expect

Comment (viceice) [reply lib/modules/datasource/cpan/index.ts:119]: it should use zod from beginning. also this causes an additional API request.

**Review by viceice — COMMENTED** [review]
see comment

**Review by viceice — COMMENTED** [review]
missing zod schema , as requested

File: lib/modules/datasource/cpan/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/modules/datasource/cpan/index.ts]: It's necessary, unless it can be done with the new URL only?

Comment (viceice) [reply lib/modules/datasource/cpan/index.ts]: should be done for every **new** request. we can leave existing as is

Comment (rarkins) [reply lib/modules/datasource/cpan/index.ts]: I meant that getting the latest tag from the new URL is necessary. Which means two requests per package is necessary unless the new URL data can obsolete the existing URL data

File: lib/modules/datasource/cpan/__fixtures__/Plack.json [hunk 3]
```diff
@@ -227,6 +237,29 @@
...
             ]
+         },
+         {
+            "_type" : "file",
+            "_source" : {
+               "deprecated" : false,
+               "maturity" : "released",
+               "module" : [
+                  {
+                     "version" : "1.0039",
```
Comment (RahulGautamSingh) [inline lib/modules/datasource/cpan/__fixtures__/Plack.json:248]:
```suggestion
                     "version" : "1.0038",
```

File: lib/modules/datasource/cpan/index.ts [hunk 4]
```diff
@@ -60,55 +60,47 @@ export class CpanDatasource extends Datasource {
...
-          if (!latestDistribution) {
-            latestDistribution = distribution;
-          }
+    let latestVersion: string | null = null;
+    if (releases) {
+      for (const release of releases) {
+        if (!latestDistribution) {
+          latestDistribution = release.distribution;
+        }
+        if (release.isLatest) {
```
Comment (RahulGautamSingh) [inline lib/modules/datasource/cpan/index.ts:86]:
```suggestion
        if (!latestVersion && release.isLatest) {
```

File: lib/modules/datasource/cpan/index.ts [hunk 5]
```diff
@@ -38,7 +38,7 @@ export class CpanDatasource extends Datasource {
     let result: ReleaseResult | null = null;
     const searchUrl = joinUrlParts(registryUrl, 'v1/file/_search');
 
-    let hits: MetaCpanApiFile[] | null = null;
+    let releases: (Release & { distribution: string })[] | null = null;
```
Comment (viceice) [inline lib/modules/datasource/cpan/index.ts:41]: use new interface

**Review by viceice — APPROVED** [review]

**Pull request merged**